torch/utils/_config_module.py

```diff
@@ -96,6 +97,29 @@ def __init__(
             ), f"justknobs only support booleans, {self.default} is not a boolean"
 
 
+if TYPE_CHECKING:
+
+    def Config(
+        default: T,
+        justknob: Optional[str] = None,
+        env_name_default: Optional[str] = None,
+        env_name_force: Optional[str] = None,
+        value_type: Optional[type] = None,
+    ) -> T:
+        ...
+
+else:
+
+    def Config(
+        default: T,
+        justknob: Optional[str] = None,
+        env_name_default: Optional[str] = None,
+        env_name_force: Optional[str] = None,
+        value_type: Optional[type] = None,
+    ) -> _Config[T]:
+        return _Config(default, justknob, env_name_default, env_name_force, value_type)
+
+
```

Comment:
It would make the machinery a little more complex but these could be collapsed if we want to avoid arg duplication:
```
if TYPE_CHECKING:
  _ConfigT: TypeAlias[T] = T
else:
  _ConfigT: TypeAlias[T] = _Config[T]

def Config(...args...) -> _ConfigT[T]:
    if TYPE_CHECKING:
        ...
    else:
        return _Config(...args....)
```